### DISCUSSION TITLE
Symbolic Regression with Gradients

### DISCUSSION THREAD

**Discussion by leonrenn** [MilesCranmer/PySR#405] [General]
Hi,

I am a huge fan of symbolic regression and of your python package. I have a question if the following would make symbolic regression better:

Say you have some data and make a neural network learn the underlying function as a black box model. 
This enables us to have gradients of that function.
Now one would be able to use the (zeroth, first, seconds, ...) derivates on which one could use SR.
Wouldn't this help validating the Regression on the regular data as one artificially creates new functions?

When having multiple outputs from the SR algorithm one could internally compare if things mathematically match and add another loss term.

Would be happy to hear your thoughts on this.

Best,

Leon

**Comment by MilesCranmer** [comment]
Hi Leon,
Thanks for your interest and the question! Yes this is definitely possible. Would https://github.com/MilesCranmer/PySR/discussions/401 provide some insight into this?

Note that there isn’t second order derivatives yet. But if/when https://github.com/MilesCranmer/SymbolicRegression.jl/pull/254 is completed, I think second order diff of expressions might be easier.

cheers!
Miles

**Reply by MilesCranmer** [reply]
And for using these in PySR, see https://astroautomata.com/PySR/examples/#9-custom-objectives

**Reply by leonrenn** [reply]
Ah ok. 
Thanks for your answer.
